extension/encoding/awslogsencodingextension/internal/unmarshaler/vpc-flow-log/const.go

```diff
@@ -0,0 +1,153 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package vpcflowlog // import "github.com/open-telemetry/opentelemetry-collector-contrib/extension/encoding/awslogsencodingextension/internal/unmarshaler/vpc-flow-log"
+
+// protocolNames are needed to know the name of the protocol number given by the field
+// protocol in a flow log record.
+// See https://www.iana.org/assignments/protocol-numbers/protocol-numbers.xhtml.
+var protocolNames = [143]string{
```

Comment:
I don't think the indices are all correct now. https://www.iana.org/assignments/protocol-numbers/protocol-numbers.xhtml says "nsh" is 145, but we only have 143 entries.

What I was suggesting before is to use this syntax:

```go
var protocolNames = [...]string{
  0: "hopopt",
  // etc.
  255: "reserved",
}
```

Reply:
Oh, you are absolutely correct :thinking:

Reply:
But then the easier way would be to have a map[int]string, don't you think? Otherwise we need empty values for numbers that do not map to any protocol name.

Reply:
I think this is a fair tradeoff when there's only 256 entries.

Reply:
I have added empty metrics. I used a unit test to compare that everything was still correct (not commited). If the protocol is ` ` (empty), then the attribute protocol name will be empty. I decided to still have the protocol attribute for those cases. What do you think?

Reply:
That's fine, but I think the syntax I used above is a little safer. You don't have to specify a value for every index that way - the unspecified ones will get an empty string. See https://go.dev/play/p/RKYse-UDfM6

Reply:
I was unaware this existed. Thanks!!